lib/packwerk/const_node_inspector.rb

```diff
@@ -24,12 +24,10 @@ def constant_name_from_node(node, ancestors:)
 
         "::" + name
       else
-        begin
-          Node.constant_name(node)
-        rescue Node::TypeError
-          nil
-        end
+        Node.constant_name(node)
       end
+    rescue Node::TypeError
+      nil
```

Comment:
This PR looks good to me, but I would like to get the approval of @tomstuart because he raised an issue [about defensive node parsing on one of my open PRs](https://github.com/Shopify/packwerk/pull/39/files#r513344732). From my understanding, this gem's design focuses on doing a good enough job uncovering dependency/privacy violations. In my opinion, defensive node parsing would fall under that because we don't fail if packwerk cannot handle the code. The downside is that we don't learn about parsing issues.

Reply:
I totally agree. I would love to rather propose a fix for the parsing errors we are running into, but am not deep enough into packwerk and parser to understand much of what's going on. 

I would be happy to pair on this or run non-defensive versions against the codebase I am looking at to give more input into what might be going on...